components/organisms/BookChildren.tsx

```diff
@@ -54,81 +51,78 @@ type Props = {
-  onItemClick(event: MouseEvent<HTMLElement>, index: ItemIndex): void;
-  onItemEditClick?(event: MouseEvent<HTMLElement>, index: ItemIndex): void;
+  onItemClick(index: ItemIndex): void;
+  onItemEditClick?(index: ItemIndex): void;
 };
 
-export default function BookChildren(props: Props) {
-  const {
-    className,
-    sections,
-    index: [sectionIndex, topicIndex],
-    isTopicEditable,
-    onItemClick,
-    onItemEditClick,
-  } = props;
+export default function BookChildren({
+  className,
+  sections,
+  index: [sectionIndex, topicIndex],
+  isTopicEditable,
+  onItemClick,
+  onItemEditClick,
+}: Props) {
   const classes = useStyles();
-  const [open, setOpen] = useState<boolean[]>(sections.map(() => true));
-  const handleItemClick = (event: MouseEvent<HTMLElement>) => {
-    const { section, topic } = event.currentTarget.dataset;
-    onItemClick(event, ([section, topic].map(Number) as unknown) as ItemIndex);
-  };
-  const handleSectionClick = (sectionItemIndex: number) => () => {
-    setOpen((open) => {
-      const newOpen = open.slice();
-      newOpen[sectionItemIndex] = !newOpen[sectionItemIndex];
-      return newOpen;
-    });
-  };
-  const handleItemEditClick = (...index: ItemIndex) => (
+  const handleItemClick = (index: ItemIndex) => () => onItemClick(index);
+  const handleSectionItemClick = (sectionItemIndex: number) => () =>
+    onItemClick([sectionItemIndex, 0]);
+  const handleItemEditClick = (index: ItemIndex) => (
     event: MouseEvent<HTMLButtonElement>
   ) => {
     event.stopPropagation();
-    onItemEditClick?.(event, index);
+    onItemEditClick?.(index);
   };
   return (
     <List disablePadding className={className}>
       {sections.map((section, sectionItemIndex) => (
-        <Section
+        <SectionItem
           key={section.id}
           section={section}
           sectionItemIndex={sectionItemIndex}
-          onSectionClick={handleSectionClick(sectionItemIndex)}
-          open={open[sectionItemIndex]}
+          end={sectionItemIndex === sections.length - 1}
+          onSectionItemClick={handleSectionItemClick(sectionItemIndex)}
         >
           {section.topics.map((topic, topicItemIndex) => (
-            <ListItem
+            <BookChildrenItem
               key={`${topic.id}:${topicItemIndex}`}
               className={clsx({
                 [classes.active]:
                   sectionIndex === sectionItemIndex &&
                   topicIndex === topicItemIndex,
               })}
+              variant="topic"
+              outlineNumber={getOutlineNumber(
+                section,
+                sectionItemIndex,
+                topicItemIndex
+              )}
+              name={topic.name}
+              end={
+                (section.topics.length > 1 &&
+                  topicItemIndex === section.topics.length - 1) ||
+                (section.topics.length === 1 &&
+                  sectionItemIndex === sections.length - 1)
+              }
```

Comment:
下記ではないですか。 `section.topics.length === 1` かつ `sectionItemIndex !== sections.length - 1` のときの振る舞いは意図通りですか。

```suggestion
              end={topicItemIndex === section.topics.length - 1}
```

Reply:
- 他のトピックも存在するtopicsのうちのトピックである、かつ、その中の最後のトピックである、場合のみ `end` が付与されること
- 他のトピックが存在しないtopicsのうちのトピックである、かつ、そのトピックをもっているセクションが最後のセクションである、場合のみ `end` が付与されること

が意図している条件です

>  `section.topics.length === 1` かつ `sectionItemIndex !== sections.length - 1` のとき

とはつまり、他のトピックが存在しないtopicsのうちのトピックである、かつ、そのトピックをもっているセクションが最後のセクションではないということなので、この場合は `end` が付与されないことを期待していました

![image](https://user-images.githubusercontent.com/9744580/125236763-76b50c00-e31f-11eb-8784-d19c5300b3d9.png)

![image](https://user-images.githubusercontent.com/9744580/125237286-66516100-e320-11eb-9cee-6e0c586b24a8.png)

Reply:
こちらも勘違いしておりました。
あえて書き直すならば下記のような疑似コードでしょうかね。

```suggestion
              end={deps ? (topicItemIndex === section.topics.length - 1) : (sectionItemIndex === sections.length - 1)}
```

Reply:
たしかにdepthプロパティに渡している条件が流用できるので変数化してこのように変更するのはよいと思いました！ https://github.com/npocccties/chibichilo/pull/459#discussion_r667647611 と平行で解決したいと思います